backend/chainlit/sql_alchemy.py

```diff
@@ -0,0 +1,449 @@
+import uuid
+import ssl
+from datetime import datetime, timezone
+import json
+from typing import Any, Dict, List, Optional, Union, TYPE_CHECKING
+import aiofiles
+import asyncio
+from dataclasses import asdict
+from sqlalchemy import text
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.ext.asyncio import create_async_engine, AsyncSession
+from sqlalchemy.orm import sessionmaker
+from azure.storage.filedatalake import FileSystemClient, ContentSettings  # type: ignore
+from chainlit.context import context
+from chainlit.logger import logger
+from chainlit.data import BaseDataLayer, queue_until_user_message
+from chainlit.user import User, PersistedUser, UserDict
+from chainlit.types import Feedback, FeedbackDict, Pagination, ThreadDict, ThreadFilter
+from literalai import PageInfo, PaginatedResponse
```

Comment:
would be good to move `PageInfo` and `PaginatedResponse` definitions to chainlit

Reply:
I 100% agree!! @willydouhard @tpatel what do you think? Honestly I'm not clear why all dataclasses aren't in `chainlit/types.py`. Classes are duplicated in both Chainlit/Literal and they really belong in Chainlit as it actually handles theses classes. Currently I can't import `chainlit/element.py` or `chainlit/step.py` because of circular import errors with chainlit/data.

Reply:
We could duplicate the definitions if needed, but we need them in the literalai sdk also. For the circular import issue, did you check https://stackoverflow.com/questions/39740632/python-type-hinting-without-cyclic-imports ?

Reply:
@willydouhard @nixent @tpatel I copy/pasted the remaining literal definitions to `backend/chainlit/types.py`. Yes I'm type checking the other necessary types using your method. Testing shows this resolves the issue, thanks!